File: sorting/LC122_best_time_to_buy_and_sell_stock_2/solution.ts
function maxProfit2(prices: number[]): number {
  let profit: number = 0;
  for (let i = 1; i < prices.length; i++) {
    if (prices[i] > prices[i-1]) {
      profit += prices[i] - prices[i-1]
    }
  }
  return profit;
}

const LC122Main = () => {
  const prices: number[] = [7, 1, 5, 3, 6, 4];
  console.log(maxProfit(prices));
}

LC122Main()

File: sorting/LC122_best_time_to_buy_and_sell_stock_2/solution.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int maxProfit(std::vector<int>& prices) {
        int profit = 0;
        for(int i = 1; i < prices.size(); i++) {
            if(prices[i] > prices[i - 1]) {
                profit += prices[i] - prices[i - 1];
            }
        }
        return profit;
    }
};

int main() {
  std::vector<int> prices = {7,1,5,3,6,4};
  Solution s;
  int output = s.maxProfit(prices);
  std::cout << output << std::endl;
}
